Java/PoolAndResourceFile/src/main/java/PoolAndResourceFile.java

```diff
@@ -0,0 +1,297 @@
+import java.util.*;
+import java.util.concurrent.TimeoutException;
+
+import com.microsoft.azure.storage.*;
+import com.microsoft.azure.storage.blob.*;
+import org.apache.commons.io.IOUtils;
+
+import com.microsoft.azure.batch.*;
+import com.microsoft.azure.batch.auth.BatchSharedKeyCredentials;
+import com.microsoft.azure.batch.protocol.models.*;
+
+public class PoolAndResourceFile {
+
+    // Create IaaS pool if pool isn't exist
+    private static CloudPool CreatePool(BatchClient client, String poolId) throws BatchErrorException, IllegalArgumentException, IOException, InterruptedException, TimeoutException {
+        // Pool only have 3 D1 VM
+        String osPublisher = "OpenLogic";
+        String osOffer = "CentOS";
+        String poolVMSize = "STANDARD_D1";
+        int poolVMCount = 1;
+        long POOL_STEADY_TIMEOUT = 5 * 60 * 1000;
+        long VM_READY_TIMEOUT = 20 * 60 * 1000;
+
+        // Check pool existing
+        if (!client.getPoolOperations().existsPool(poolId)) {
+
+            // Get the sku image reference
+            List<NodeAgentSku> skus = client.getAccountOperations().listNodeAgentSkus();
+            String skuId = null;
+            ImageReference imageRef = null;
+
+            for (NodeAgentSku sku : skus) {
+                if (sku.getOsType() == OSType.LINUX) {
+                    for (ImageReference imgRef : sku.getVerifiedImageReferences()) {
+                        if (imgRef.getPublisher().equalsIgnoreCase(osPublisher) && imgRef.getOffer().equalsIgnoreCase(osOffer)) {
+                            imageRef = imgRef;
+                            skuId = sku.getId();
+                            break;
+                        }
+                    }
+                }
+            }
+
+            // Use IaaS VM with Linux
+            VirtualMachineConfiguration configuration = new VirtualMachineConfiguration();
+            configuration.setNodeAgentSKUId(skuId);
+            configuration.setImageReference(imageRef);
+
+            client.getPoolOperations().createPool(poolId, poolVMSize, configuration, poolVMCount);
+        }
+
+        long startTime = System.currentTimeMillis();
+        long elapsedTime = 0L;
+        boolean steady = false;
+
+        // Let's wait max 5 minute for VM to be allocated
+        while (elapsedTime < POOL_STEADY_TIMEOUT) {
+            CloudPool pool = client.getPoolOperations().getPool(poolId);
+            if (pool.getAllocationState() == AllocationState.STEADY) {
+                steady = true;
+                break;
+            }
+            System.out.println("wait 30 seconds for pool steady...");
+            Thread.sleep(30 * 1000);
+            elapsedTime = (new Date()).getTime() - startTime;
+        }
+
+        if (!steady) {
+            throw new TimeoutException("Pool wasn't steady at time");
+        }
+
+        // VMs of the pool don't need to be IDLE state in order to submit job
+        // The following code just an example to check VM state
```

Comment:
nitpick: "The VMs in the pool don't need to be in and IDLE state in order to submit a job. The following code is just an example of how to poll for the VM state"